src/pages/place/[placeId]/status.js

```diff
@@ -1,55 +1,176 @@
 import { useRouter } from 'next/router';
 import { QueryStatus } from '../../../hooks/api/authenticated-query';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { getApplicationServerSideProps } from '../../../utils/getApplicationServerSideProps';
 import { Alert, AlertVariants } from '../../../components/publiq-ui/Alert';
 import { Button, ButtonVariants } from '../../../components/publiq-ui/Button';
 import { Spinner } from '../../../components/publiq-ui/Spinner';
+import { RadioButtonGroup } from '../../../components/publiq-ui/RadioButtonGroup';
 import { Page } from '../../../components/publiq-ui/Page';
+import { TextAreaWithLabel } from '../../../components/publiq-ui/TextAreaWithLabel';
 import { useTranslation } from 'react-i18next';
-import { useGetPlaceById } from '../../../hooks/api/places';
+import { useChangeStatus, useGetPlaceById } from '../../../hooks/api/places';
+import { Text } from '../../../components/publiq-ui/Text';
+import { getValueFromTheme } from '../../../components/publiq-ui/theme';
+import { Stack } from '../../../components/publiq-ui/Stack';
+import { Inline } from '../../../components/publiq-ui/Inline';
+import { dehydrate } from 'react-query/hydration';
+
+const getValue = getValueFromTheme('statusPage');
+
+const OfferStatus = {
+  AVAILABLE: 'Available',
+  TEMPORARILY_UNAVAILABLE: 'TemporarilyUnavailable',
+  UNAVAILABLE: 'Unavailable',
+};
+
+const maxLengthReason = 200;
 
 const Status = () => {
   const { t, i18n } = useTranslation();
   const router = useRouter();
   const { placeId } = router.query;
 
-  const [errorMessage, setErrorMessage] = useState();
+  const [type, setType] = useState('');
+  const [reason, setReason] = useState('');
+
+  const handleSuccessChangeStatus = () =>
+    router.push(`/place/${placeId}/preview`);
+
+  const {
+    data: place = {},
+    status: getPlaceByIdStatus,
+    error: getPlaceByIdError,
+  } = useGetPlaceById({ id: placeId });
+
+  const {
+    mutate: changeStatus,
+    status: changeStatusStatus,
+    error: changeStatusError,
+  } = useChangeStatus({
+    onSuccess: handleSuccessChangeStatus,
+  });
 
-  const handleError = (error) => {
-    setErrorMessage(error.message);
-  };
-  const { data: place = {}, status } = useGetPlaceById(
-    { id: placeId },
-    { onError: handleError },
-  );
   const name = place.name?.[i18n.language] ?? place.name?.[place.mainLanguage];
+  const rawStatusType = place?.status?.type;
```

Comment:
What do you mean by `raw`?

Reply:
as in `raw data from the api query`, these values get set as state values in the component.
e.g. type gets set to the value of rawStatusType

Reply:
Ok, clear. Thanks for explaining. Is `original` then a better prefix?

Reply:
I'm going to keep the raw naming for now because we use this in several other places